packages/components/src/internal/components/editable/actions.ts

```diff
@@ -1450,68 +1264,38 @@ interface ParseLookupPayload {
     values: List<ValueDescriptor>;
 }
 
-function isReadonlyRow(row: Map<string, any>, pkCols: QueryColumn[], readonlyRows: string[]): boolean {
-    if (pkCols.length === 1 && row) {
-        // Coerce pkValue to string because it may actually be a number or other type, and readonlyRows is string[]
-        const pkValue = caseInsensitive(row.toJS(), pkCols[0].fieldKey)?.toString();
-        return readonlyRows.includes(pkValue);
-    }
-
-    return false;
-}
-
 async function insertPastedData(
-    dataKeys: List<any>,
-    data: Map<any, Map<string, any>>,
-    queryInfo: QueryInfo,
-    columns: QueryColumn[],
     editorModel: EditorModel,
     paste: PasteModel,
-    columnMetadata: Map<string, EditableColumnMetadata>,
     readonlyRows: string[],
-    lockedRows: string[],
     lockRowCount: boolean,
     forUpdate: boolean,
     targetContainerPath: string,
     selectCells: boolean
-): Promise<EditorModelAndGridData> {
+): Promise<Partial<EditorModel>> {
     const pastedData = paste.payload.data;
     let cellMessages = editorModel.cellMessages;
     let cellValues = editorModel.cellValues;
     const selectionCells: string[] = [];
     let rowCount = editorModel.rowCount;
-    let updatedDataKeys: List<any>;
-    let updatedData: Map<any, Map<string, any>>;
 
     if (paste.rowsToAdd > 0 && !lockRowCount) {
         rowCount += paste.rowsToAdd;
-        const dataChanges = addRowsToGridData(dataKeys, data, paste.rowsToAdd);
-        updatedData = dataChanges.data;
-        updatedDataKeys = dataChanges.dataKeys;
     }
 
     const byColumnValues = getPasteValuesByColumn(paste);
     const lookupColumnContainerCache = {};
+    // TODO: colMin may be wrong here because we're now storing hidden columns (e.g. RowId) in EditorModel
```

Comment:
we are storing hidden columns in the EditorModel, but those aren't in the orderedColumns info right? So I would assume that the colMin, colMax, etc. here are based on the orderedColumns?

Reply:
Yes, I believe this is totally safe, I was just being paranoid and writing these TODOs so that if I hit an test failure I could just look at all the places I left these notes to myself and hopefully track down the potential issues faster.